CDTDatastore/query/CDTQIndexManager.m

```diff
@@ -262,14 +263,59 @@ - (NSString *)ensureIndexed:(NSArray * /* NSString */)fieldNames
                        type:(NSString *)type
                    settings:(NSDictionary *)indexSettings
 {
+    CDTQIndexType indexType = [CDTQIndexManager indexTypeForString:type];
+    return
+        [self ensureIndexed:fieldNames withName:indexName ofType:indexType settings:indexSettings];
+}
+
+#pragma mark New methods
+
+- (NSString *)ensureIndexed:(NSArray<NSString *> *)fieldNames
+                   withName:(NSString *)indexName
+                     ofType:(CDTQIndexType)type
+{
+    return [self ensureIndexed:fieldNames withName:indexName ofType:type settings:nil];
+}
+
+- (NSString *)ensureIndexed:(NSArray<NSString *> *)fieldNames
+                   withName:(NSString *)indexName
+                     ofType:(CDTQIndexType)type
+                   settings:(NSDictionary *)indexSettings
+{
     return [CDTQIndexCreator ensureIndexed:[CDTQIndex index:indexName
                                                  withFields:fieldNames
-                                                     ofType:type
+                                                       type:type
                                                withSettings:indexSettings]
                                 inDatabase:_database
                              fromDatastore:_datastore];
 }
 
++ (CDTQIndexType)indexTypeForString:(NSString *)string
+{
+    if ([string isEqualToString:@"text"]) {
+        return CDTQIndexTypeText;
+    } else if ([string isEqualToString:@"json"]) {
+        return CDTQIndexTypeJSON;
+    } else {
+        @throw [NSException exceptionWithName:@"InvalidIndexException"
+                                       reason:@"Index type provided is not a valid index type."
+                                     userInfo:@{
+                                         @"Expected" : @"text or json",
+                                         @"Actual" : string
+                                     }];
+    }
+}
+
++ (NSString *)stringForIndexType:(CDTQIndexType)indexType /* __attribute__((const))*/
+{
+    switch (indexType) {
+        case CDTQIndexTypeText:
+            return @"text";
+        case CDTQIndexTypeJSON:
+            return @"json";
+    }
```

Comment:
Does this need a default? That is, can I fake out an enum with some trickery like `[... stringForIndexType:(CDTQIndexType)3]`? If I can we probably need protections elsewhere too.